DatadogTrace/Sources/DatadogTracer.swift

```diff
@@ -110,12 +112,14 @@ internal class DatadogTracer: OTTracer {
 
     // MARK: - Internal
 
-    internal func createSpanContext(parentSpanContext: DDSpanContext? = nil) -> DDSpanContext {
+    internal func createSpanContext(parentSpanContext: DDSpanContext?, using sampler: Sampler) -> DDSpanContext {
         return DDSpanContext(
             traceID: parentSpanContext?.traceID ?? traceIDGenerator.generate(),
             spanID: spanIDGenerator.generate(),
             parentSpanID: parentSpanContext?.spanID,
-            baggageItems: BaggageItems(parent: parentSpanContext?.baggageItems)
+            baggageItems: BaggageItems(parent: parentSpanContext?.baggageItems),
+            sampleRate: parentSpanContext?.sampleRate ?? sampler.samplingRate,
+            isKept: parentSpanContext?.isKept ?? sampler.sample()
```

Comment:
💡 This is the main point for head-based sampling implementation in this PR:
- use the sampling decision from the parent span (if available)
- or determine it based on provided sampler (which can be local or distributed trace sampler).